balancer/rls/balancer.go

```diff
@@ -326,6 +340,7 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 		// `resizeCache` boolean) because `cacheMu` needs to be grabbed before
 		// `stateMu` if we are to hold both locks at the same time.
 		b.cacheMu.Lock()
+		b.dataCache.updateRLSServerTarget(newCfg.lookupService)
```

Comment:
This is under `if resizeCache`. That doesn't feel right to me. The RLS server target could change without the cache size changing.

Reply:
Oh hmmm yeah good point.